### PR TITLE
Trans Updates - Replace,New Stack, and Sequence

### LINKED ISSUE #2067: [FEATURE MODIFICATION] Trans::Replace A method of changing the entire game state stack easily
## Feature

amethyst::Trans

## Change

Add in a Trans::Replace(new_state) which removes each state on the stack ( and calls the on_stop command for each) and replaces in with new_state.

## Reason

To simplify the use case where you want to switch primary states from a pushed state. Example: you want to be able to return to the main game menu from a paused state that is pushed on top of the main game state.

## Impact

Nothing else

### PR DESCRIPTION
**Pull request by mrhatman**
## Description

Added new values to the Trans Enum for Transitioning States.

Closes #2067 

## Additions

- Trans::Replace(new_state): Delete the stack and replace with the new state.

- Trans::Sequence(Trans[]): Execute a sequence of transitions.

- Trans::NewState(New_States[]): Remove the existing stack and replace with New_States.

## PR Checklist

By placing an x in the boxes I certify that I have:

- [ ] Updated the content of the book if this PR would make the book outdated.
- [x] Added a changelog entry if this will impact users, or modified more than 5 lines of Rust that wasn't a doc comment.
- [x] Added unit tests for new code added in this PR.
- [x] Acknowledged that by making this pull request I release this code under an MIT/Apache 2.0 dual licensing scheme.

If this modified or created any rs files:

- [x] Ran `cargo +stable fmt --all`
- [x] Ran `cargo clippy --all --features "empty"`
- [x] Ran `cargo test --all --features "empty"`

### CODE REVIEW

**Comment by mrhatman** [comment]
Can I get a pipeline retry?

File: src/state.rs [hunk 1]
```diff
@@ -519,6 +539,49 @@ impl<'a, T, E: Send + Sync + 'static> StateMachine<'a, T, E> {
...
+            //State was just pushed, thus pop will always succeed
+            let new_state = self.state_stack.last_mut().unwrap();
+            new_state.on_start(StateData { world, data });
+        }
+    }
+
+    /// Removes all states from the stack and replaces it with a new stack.
+    pub(crate) fn new_stack(&mut self, states: Vec<Box<dyn State<T, E>>>, data: StateData<'_, T>) {
+        if self.running {
+            //revome all current states
```
Comment (jojolepro) [inline src/state.rs]:
```suggestion
            //remove all current states
```

**Comment by jojolepro** [comment]
@mrhatman we can merge once you apply my suggested change for the typo error :)

**Comment by jojolepro** [comment]
bors r+

Thanks a lot!

**Pull request merged**